golang/cosmos/x/vibc/types/receiver.go

```diff
@@ -121,21 +122,20 @@ func (ir Receiver) Receive(cctx context.Context, jsonRequest string) (jsonReply
 			timeoutTimestamp = uint64(ctx.BlockTime().UnixNano()) + msg.RelativeTimeoutNs
 		}
 
-		packet := channeltypes.NewPacket(
-			msg.Packet.Data, 0,
-			msg.Packet.SourcePort, msg.Packet.SourceChannel,
-			msg.Packet.DestinationPort, msg.Packet.DestinationChannel,
-			msg.Packet.TimeoutHeight, timeoutTimestamp,
-		)
+		packet := types.CopyToIBCPacket(msg.Packet)
+		packet.TimeoutTimestamp = timeoutTimestamp
 		seq, err := impl.ReceiveSendPacket(ctx, packet)
```

Comment:
I was surprised that `ReceiveSendPacket` accepted an IBCPacket but makes sense since it's compatible with `PacketI`